modules/node_modules/@ncigdc/dux/cart.js

```diff
@@ -6,56 +6,161 @@ import { REHYDRATE } from 'redux-persist/constants';
+    const incomingFileArray = incomingFile.constructor === Array ? incomingFile : [incomingFile];
     const existingFiles = getState().cart.files;
-    const nextFiles = _.xorBy(existingFiles, incomingFilesArray, 'file_id');
+    const nextFiles = _.xorBy(existingFiles, incomingFileArray, 'file_id');
 
-    if (nextFiles.length === existingFiles.length + 1) {
+    if (nextFiles.length > existingFiles.length) {
+      dispatch(notify(getNotificationComponent('add', `add/${incomingFile.file_name}`, {
+        action: 'Added',
+        file: incomingFile.file_name,
+        prapositon: 'to',
+        undo: {
+          files: incomingFile
+        }
+      }, dispatch)));
+    }
+
+    if (nextFiles.length < existingFiles.length) {
+      dispatch(notify(getNotificationComponent('remove', `remove/${incomingFile.file_name}`, {
+        action: 'Removed',
+        file: incomingFile.file_name || incomingFileArray.length,
+        prapositon: 'from',
+        undo: {
+          files: incomingFile
+        }
+      }, dispatch)));
+    }
+
+    dispatch({
+      type: UPDATE_CART,
+      payload: nextFiles,
+    });
+  };
+}
+
+function fetchFiles (currentFilters, total) {
+  return async dispatch => {
+    if (total <= 10000){
+      const search = stringify({
+        filters: JSON.stringify(currentFilters),
+        size: total,
+        fields: 'access,file_id,file_size,cases.project.project_id'
+      });
+      const { data } = await fetchApi(`files?${search}`);
+      dispatch(addAllFilesInCart(data.hits))
+    } else {
       dispatch(notify({
         action: 'add',
-        id: `add/${incomingFiles.file_name}`,
         component:
-  <Column>
-    <span>Added <strong>{incomingFiles.file_name}</strong> to the cart.</span>
-    <span style={center}>
-      <UndoIcon />
-      <a onClick={() => dispatch(toggleInCart(incomingFiles))}>Undo</a>
-    </span>
-  </Column>,
+          <NotificationColumn>
+            <span>The cart is limited to 10,000 files. Please narrow down your search criteria to be able to add files to your cart.</span>
+          </NotificationColumn>,
       }));
     }
+  }
+}
+
+function addAllFilesInCart(incomingFiles) {
+  return (dispatch, getState) => {
+    const incomingFilesArray = incomingFiles.constructor === Array ? incomingFiles : [incomingFiles];
+    const existingFiles = getState().cart.files;
+    const nextFiles = _.filter(incomingFilesArray, (file) => !_.some(existingFiles, {'file_id':file.file_id}));
+    const filesInCart = incomingFilesArray.length - nextFiles.length;
+
+    if(nextFiles && nextFiles.length < incomingFilesArray.length){
+      dispatch(notify(getNotificationComponent('add', `add/some`, {
+        action: 'Added',
+        file: nextFiles.length,
+        fileText: nextFiles.length>1 ? 'files ' : 'file ',
+        extraText: `<strong>${filesInCart}</strong> ${filesInCart>1 ? 'files' : 'file'} already in cart, not added.`,
```

Comment:
use jsx instead of a string

``` 
(
<span>
  <strong>{filesInCart}</strong> {filesInCart > 1 ? 'files' : 'file'} already in cart, not added.
</span>
),